compiler/incremental-compilation-impl/src/org/jetbrains/kotlin/incremental/classpathDiff/ClasspathSnapshot.kt

```diff
@@ -36,7 +38,14 @@ class ClasspathEntrySnapshot(
  * `KotlinCompile` task and the task needs to support compile avoidance. For example, this class should contain public method signatures,
  * and should not contain private method signatures, or method implementations.
  */
-sealed class ClassSnapshot
+sealed class ClassSnapshot {
+
+    /** Computes the hash of this [ClassSnapshot] and returns a [ClassSnapshotWithHash]. */
+    fun addHash() = ClassSnapshotWithHash(this, ClassSnapshotExternalizer.toByteArray(this).md5())
```

Comment:
I'm not sure how to optimize this further because the serialized ByteArray is the complete representation of the object, so we have to go through it.

Yes, the hash function can be changed but it's true everywhere else, so maybe it's not worth adding a comment?

Reply:
yeah, because we any how have custom serialization I guess it is fine, but keeping everything in a byte array is not really needed, right? you can update the hash on the fly with every property read.

Reply:
Okay, I'll keep that in mind and we can return to this if it shows up as a problem during profiling.